types/openid-client/index.d.ts

```diff
@@ -8,6 +8,18 @@
 /// <reference types="node" />
 
 import { IncomingMessage } from 'http';
+import { GotOptions } from 'got';
+
+export { }; // Disable automatic export of all module members (make it explicit)
```

Comment:
Sorry, what’s this? Given that other things are already exported, I’m pretty sure this isn’t needed?

Reply:
Hi @andrewbranch 

You mean the `export { }`? 

This is to avoid exporting _all_ declarations (instead require an explicit `export`). Especially, I want to avoid that `HttpRequestOptions` and `CustomHttpOptionsProvider` get exported as these are "internal" types.

BTW: The DT linting rules recommended to do this:

![image](https://user-images.githubusercontent.com/388796/60831465-f13f7400-a1b9-11e9-9dd0-2aeeb3ed3bfe.png)

Reply:
Out of curiosity, why wouldn't you want someone to access these definitions?

Reply:
My thoughts:
1. These types are just utility types / type aliases to avoid repetition/duplication of them in the following `Issuer` and `Client` types. Both types don't exist as _explicit_ concept in the "openid-client" library.
1. These types unnecessarily pollute the `import { ... } from 'openid-client'` "namespace" (i.e. produce unnecessary items in the syntax completion).

.. that said, I'm not totally against exposing them, so if you think it's better to always export all interfaces, including "intermediate" types, I'lll change that.

Reply:
Any more comments @andrewbranch and @rbuckton and is it good for merging?

Reply:
No issue from me 👍